File: lc_go/lcGo/array/maximumPossibleSize3523.go
package main

import "fmt"

func maximumPossibleSize(nums []int) (ans int) {
	mx := 0
	for _, x := range nums {
		if x >= mx {
			mx = x
			ans++
		}
	}
	return
}

func main() {
	nums := []int{4, 2, 5, 3, 5}
	res := maximumPossibleSize(nums)
	fmt.Println(res)
}

File: lc_cpp/lcCpp/arraY/maximumPossibleSize3523.cpp
//
// Created by 24276 on 2025/4/21.
//
#include <vector>
#include <iostream>
using namespace std;
class maximum_possible_size3523 {
public:
    int maximumPossibleSize(vector<int>& nums) {
        int ans = 0, mx = 0;
        for (int x : nums) {
            if (x >= mx) {
                mx = x;
                ans++;
            }
        }
        return ans;
    }
};

int main() {
    maximum_possible_size3523 sol;
    vector<int> nums = {4,2,5,3,5};
    cout << sol.maximumPossibleSize(nums);
    return 0;
}
